### DISCUSSION TITLE
Is there a way to divide/multiply pixels by floating point number?

### DISCUSSION THREAD

**Discussion by a-gavrilkov** [sn4k3/UVtools#575] [Q&A]
Hi! I need to reduce the pixel brightness of the model's surface to 75% but It seems that it is not possible to do with Pixel Arithmetics -> Multiply/Divide with brightness since it only accepts integer numbers. Is there a way to do this using scripting?

**Comment by sn4k3** [comment]
Pixel data type is **byte (0-255)** and that tool use Mat to Mat operations which require same data type or it will trigger an error.
However via scripting and if you iterate pixels you can do regular math, like: `pixel = (byte)(pixel * 0.75)` i'm truncating but you can opt for floor, ceil or round.

To simplify this, and I guess your model is all solid white [255] (beside AA) you can subtract **64** (75% of 255) to the model pixels to obtain the "same" effect.

```
255 * 0.75 = 191,25
255 - 64 = 191
```

Of course this will not be the same if you got fade areas and may mess with your AA, but if your model is solid white then the result is the same.

As alternative and best way instead of subtracting you can set pixel directly to **191** to solve the AA problem or pixels fader than **255**
